storey/targets.py

```diff
@@ -686,9 +691,11 @@ async def _worker(self):
             await self._storage.close()
 
     async def _lazy_init(self):
-        if not self._shard_count:
+        if not self._initialized:
+            await self._storage._create_stream(self._container, self._stream_path, self._shard_count,
+                                               self._retention_period_hours)
+            # get actual number of shards (for pre existing stream)
             response = await self._storage._describe(self._container, self._stream_path)
```

Comment:
Not a big deal, but this is only necessary when `create_stream` failed with 409.